### ISSUE TITLE
[release-1.21] Allow customizing of the static pod manifests for CPU, files, and directories

### ISSUE THREAD

**Issue by Oats87** [rancher/rke2#1525]
Fix for **Allow customizing of the static pod manifests for CPU, files, and directories** on branch **release-1.21**
* https://github.com/rancher/rke2/issues/1233

**Comment by rancher-max** [comment]
#### Validated the flags are present on v1.21.3-rc5+rke2r2

Made the following confirmations with set flags, using config.yaml shown below:
```yaml
write-kubeconfig-mode: 644
control-plane-resource-requests: kube-apiserver-cpu=250m,kube-apiserver-memory=256M,kube-scheduler-cpu=125m,kube-scheduler-memory=256M,etcd-cpu=500m
control-plane-resource-limits: kube-apiserver-cpu=500m,kube-apiserver-memory=512M,kube-scheduler-cpu=250m,kube-scheduler-memory=512M,etcd-cpu=1000m
kube-apiserver-extra-mount: /root/test/ka:/max/is/testing
kube-scheduler-extra-mount: /root/test/ks:/max/is/testing
kube-controller-manager-extra-mount: /root/test/kc:/max/is/testing
kube-proxy-extra-mount: /root/test/kp:/max/is/testing
etcd-extra-mount: /root/test/e:/max/is/testing:ro
cloud-controller-manager-extra-mount: 
  - /root/test/ccm:/max/is/testing
  - /root/test/ccmtwo:/max/is/boop:ro
kube-apiserver-extra-env: 
  - FOO=BAR
  - ANOTHER=OTHER
kube-scheduler-extra-env: BAR=BAZ
kube-controller-manager-extra-env: BAZ=FOO
kube-proxy-extra-env: HI=BYE
etcd-extra-env: GIMME=ALLTHETHINGS
cloud-controller-manager-extra-env: BOOP=BEEP
```

```
$ k describe -n kube-system pod/cloud-controller-manager-maxt2
...
    Environment:
      FILE_HASH:  330f44e7d71e959463fcbb6a7ca875b282e1f5d2c614f9c6b4b835642947a590
      NO_PROXY:   .svc,.cluster.local,10.42.0.0/16,10.43.0.0/16
      BOOP:       BEEP
    Mounts:
      /etc/ca-certificates from dir1 (rw)
      /etc/ssl/certs from dir0 (rw)
      /max/is/boop from extra-mount-1 (ro)
      /max/is/testing from extra-mount-0 (rw)
...

$ k exec -it -n kube-system pod/cloud-controller-manager-maxt2 -- /bin/bash

bash-4.2# echo "hi" > /max/is/boop/other.txt   
bash: /max/is/boop/other.txt: Read-only file system

bash-4.2# echo "hi" > /max/is/testing/other.txt
bash-4.2# cat /max/is/testing/other.txt 
hi
bash-4.2# cat /max/is/boop/test.txt 
ccmtwo-test
bash-4.2# cat /max/is/testing/test.txt 
ccm-test

bash-4.2# env
...
BOOP=BEEP
...
```

Also confirmed pods like `kube-apiserver` where limits were set contained the correct info:
```
Limits:
  cpu:     500m
  memory:  512M
Requests:
  cpu:     250m
  memory:  256M
```

Also confirmed all pods up and running successfully

**Comment by rancher-max** [comment]
I didn't mean to close this. @Oats87 This is working for everything EXCEPT `kube-proxy` pod. I don't see the Mounts or Env set in this static pod but I do see it in the other ones.

**Comment by brandond** [comment]
kubeproxy PRs have been merged, moving back to to-test

**Comment by cjellick** [comment]
My expectation is that this goes into 1.21.3+rke2r2, but does not get exposed in KDM for 2.6.0

**Comment by rancher-max** [comment]
Validated remaining issue is fixed in v1.21.3-rc6+rke2r2
